File: Data Structures & Algorithms/length-of-last-word/submission-2.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        if(s.length() == 1 && s[0] != ' '){
            return 1;
        }
        int start = -1 , end = -1;
        int i = s.length() -1;
        while(i >= 0 && s[i] == ' '){
            i-=1;
        }
        end = i;
        while(i >= 0 && s[i] != ' '){
            i-=1;
        }
        start = i;
        return end - start ;
    }
};

File: Data Structures & Algorithms/rotate-array/submission-2.java
class Solution {
    public void rotate(int[] nums, int k) {
        int moveLen = k % nums.length;
        if(moveLen == 0 || nums.length <= 1){
            return;
        }
        
        int c = 0;
        int start = 0;

        while(c < nums.length){
            int index = start;
            int temp = nums[index];
            
            while(true) {
                int iMove = (index + moveLen) % nums.length;
                
                int temp2 = nums[iMove];
                nums[iMove] = temp;
                temp = temp2;
                
                index = iMove;
                c += 1;
                
                if(index == start){
                    break; 
                }
            }
            start++; 
        }
    }
}